src/views/forms/AccountManagementForms/DepositForm.tsx

```diff
@@ -135,10 +161,15 @@ export const DepositForm = ({ onDeposit, onError }: DepositFormProps) => {
   }, [debouncedAmountBN.toNumber()]);
 
   useEffect(() => {
-    if (dydxAddress && evmAddress) {
+    if (dydxAddress) {
       // TODO: this is for fixing a race condition where the sourceAddress is not set in time.
       // worth investigating a better fix on abacus
-      abacusStateManager.setTransfersSourceAddress(evmAddress);
+      if (evmAddress) {
+        abacusStateManager.setTransfersSourceAddress(evmAddress);
+      }
+      if (walletType === WalletType.Keplr && nobleAddress) {
+        abacusStateManager.setTransfersSourceAddress(nobleAddress);
+      }
```

Comment:
When connected to the Keplr wallet, set the sourceAddress to nobleAddress since evmAddress is not available.